src/game/game/game.c

```diff
@@ -159,16 +159,13 @@ bool Game_Start(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
         break;
 
     case GFL_SELECT:
-        // reset current info to the defaults so that we do not do
-        // Item_GlobalReplace in the inventory initialization routines too early
         Savegame_InitCurrentInfo();
-        Savegame_LoadOnlyResumeInfo(g_GameInfo.current_save_slot, &g_GameInfo);
-        for (int i = level_num; i < g_GameFlow.level_count; i++) {
-            Savegame_ResetCurrentInfo(i);
-        }
-        if (level_num <= g_GameFlow.first_level_num) {
-            Savegame_InitCurrentInfo();
-        } else {
+        if (level_num > g_GameFlow.first_level_num) {
+            Savegame_LoadOnlyResumeInfo(
+                g_GameInfo.current_save_slot, &g_GameInfo);
+            for (int i = level_num; i < g_GameFlow.level_count; i++) {
+                Savegame_ResetCurrentInfo(i);
+            }
```

Comment:
Do I understand it right that this change technically shouldn't do anything and is cosmetic, since `Savegame_InitCurrentInfo` executed the second time for levels <= the first level would completely override the results of `Savegame_LoadOnlyResumeInfo` and `Savegame_ResetCurrentInfo`?

Reply:
Yes cosmetic. Basically the comment was copied and pasted from the `GFL_SAVED` case and wasn't reflecting what was actually happening. In the select level / restart cases, the level is loaded after the current info was restored (since those processes don't call an inventory init like the load game process does).

And yes you are spot on with the second part of your sentence. I did some testing with `remove_guns` and `remove_scions` to make sure that Lara started the level with the proper items even after a restart, level select, save after picking up a weapon, etc.